### PR TITLE
Websocket onclose api consist with Web API

### PR DESCRIPTION
**Pull request by novain1102**
Re: https://github.com/cocos/cocos-engine/issues/12286#

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/network/WebSocket-apple.mm [hunk 1]
```diff
@@ -177,8 +177,12 @@ - (void)webSocket:(SRWebSocket *)webSocket didReceiveMessageWithData:(NSData *)n
 }
 
 - (void)webSocket:(SRWebSocket *)webSocket didCloseWithCode:(NSInteger)code reason:(NSString *)reason wasClean:(BOOL)wasClean {
-    if (!_isDestroyed)
-        _delegate->onClose(_ccws);
+    if (!_isDestroyed) {
+        auto codeArg = static_cast<uint16_t>(code);
+        const ccstd::string& reasonArg = reason == nil ? "no_resaon" : ccstd::string([reason UTF8String]);
```
Comment (minggo) [inline native/cocos/network/WebSocket-apple.mm]: Use reference for temple string?

Comment (novain1102) [reply native/cocos/network/WebSocket-apple.mm]: fixed.

File: native/cocos/network/WebSocket-libwebsockets.cpp [hunk 2]
```diff
@@ -234,7 +234,7 @@ class WebSocketImpl {
     int onClientReceivedData(void *in, ssize_t len);
     int onConnectionOpened();
     int onConnectionError();
-    int onConnectionClosed();
+    int onConnectionClosed(uint16_t code, const ccstd::string& reason);
```
Comment (PatriceJiang) [inline native/cocos/network/WebSocket-libwebsockets.cpp:237]: Why not use `int` for parameter `code`?

Comment (novain1102) [reply native/cocos/network/WebSocket-libwebsockets.cpp:237]: https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent said that `code` is an `unsigned short`

**Review by PatriceJiang — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**